static/app/components/events/interfaces/spans/scrollbarManager.tsx

```diff
@@ -473,6 +468,49 @@ export class Provider extends Component<Props, State> {
     }
   };
 
+  markSpanOutOfView = (spanId: string) => {
+    if (!this.spansInView.removeSpan(spanId)) {
+      return;
+    }
+
+    const left = this.spansInView.getScrollVal();
+    this.throttledScroll(left, true);
+  };
+
+  markSpanInView = (spanId: string, treeDepth: number) => {
+    if (!this.spansInView.addSpan(spanId, treeDepth)) {
+      return;
+    }
+
+    const left = this.spansInView.getScrollVal();
+    this.throttledScroll(left, true);
+  };
+
+  startAnimation() {
+    selectRefs(this.contentSpanBar, (spanBarDOM: HTMLDivElement) => {
+      spanBarDOM.style.transition = 'transform 0.3s';
+    });
+
+    if (this.animationTimeout) {
+      clearTimeout(this.animationTimeout);
+    }
+
+    // This timeout is set to trigger immediately after the animation ends, to disable the animation.
+    // The animation needs to be cleared, otherwise manual horizontal scrolling will be animated
+    this.animationTimeout = setTimeout(() => {
+      selectRefs(this.contentSpanBar, (spanBarDOM: HTMLDivElement) => {
+        spanBarDOM.style.transition = '';
+      });
+      this.animationTimeout = null;
+    }, 300);
+  }
+
+  disableAnimation() {
+    selectRefs(this.contentSpanBar, (spanBarDOM: HTMLDivElement) => {
+      spanBarDOM.style.transition = '';
+    });
+  }
```

Comment:
I don't think you need any of this, just put a static `transition: transform 0.3s;` on the element style wherever it is defined, it's okay if it's css animating during wheel movements, it won't look that weird I think.